auth/src/main/java/com/firebase/ui/auth/util/data/PhoneNumberUtils.java

```diff
@@ -1362,13 +183,259 @@ private static String stripPlusSign(String phoneNumber) {
+        map.put(678, singletonList("VU"));
+        map.put(679, singletonList("FJ"));
+        map.put(680, singletonList("PW"));
+        map.put(681, singletonList("WF"));
+        map.put(682, singletonList("CK"));
+        map.put(683, singletonList("NU"));
+        map.put(685, singletonList("WS"));
+        map.put(686, singletonList("KI"));
+        map.put(687, singletonList("NC"));
+        map.put(688, singletonList("TV"));
+        map.put(689, singletonList("PF"));
+        map.put(690, singletonList("TK"));
+        map.put(691, singletonList("FM"));
+        map.put(692, singletonList("MH"));
+        map.put(800, singletonList("001"));
+        map.put(808, singletonList("001"));
+        map.put(850, singletonList("KP"));
+        map.put(852, singletonList("HK"));
+        map.put(853, singletonList("MO"));
+        map.put(855, singletonList("KH"));
+        map.put(856, singletonList("LA"));
+        map.put(870, singletonList("001"));
+        map.put(878, singletonList("001"));
+        map.put(880, singletonList("BD"));
+        map.put(881, singletonList("001"));
+        map.put(882, singletonList("001"));
+        map.put(883, singletonList("001"));
+        map.put(886, singletonList("TW"));
+        map.put(888, singletonList("001"));
+        map.put(960, singletonList("MV"));
+        map.put(961, singletonList("LB"));
+        map.put(962, singletonList("JO"));
+        map.put(963, singletonList("SY"));
+        map.put(964, singletonList("IQ"));
+        map.put(965, singletonList("KW"));
+        map.put(966, singletonList("SA"));
+        map.put(967, singletonList("YE"));
+        map.put(968, singletonList("OM"));
+        map.put(970, singletonList("PS"));
+        map.put(971, singletonList("AE"));
+        map.put(972, singletonList("IL"));
+        map.put(973, singletonList("BH"));
+        map.put(974, singletonList("QA"));
+        map.put(975, singletonList("BT"));
+        map.put(976, singletonList("MN"));
+        map.put(977, singletonList("NP"));
+        map.put(979, singletonList("001"));
+        map.put(992, singletonList("TJ"));
+        map.put(993, singletonList("TM"));
+        map.put(994, singletonList("AZ"));
+        map.put(995, singletonList("GE"));
+        map.put(996, singletonList("KG"));
+        map.put(998, singletonList("UZ"));
+
+        return map;
+    }
+
+    private static Map<String, Integer> createCountryCodeByIsoMap() {
+        Map<String, Integer> map = new HashMap<>(MAX_COUNTRIES);
+
+        for (int i = 0; i < COUNTRY_TO_REGION_CODES.size(); i++) {
+            int code = COUNTRY_TO_REGION_CODES.keyAt(i);
+            List<String> regions = COUNTRY_TO_REGION_CODES.get(code);
+
+            for (String region : regions) {
+                map.put(region, code);
+            }
+        }
+
+        // TODO Figure out why these exceptions exist.
+        // This map used to be hardcoded so this is the diff from the generated version.
+        map.remove("001");
```

Comment:
Ok so it looks like in `COUNTRY_TO_REGION_CODES` we have:

```java
map.put(800, singletonList("001"));
map.put(808, singletonList("001"));
// ...
map.put(870, singletonList("001"));
map.put(878, singletonList("001"));
// ...
map.put(881, singletonList("001"));
map.put(882, singletonList("001"));
map.put(883, singletonList("001"));
// ...
map.put(888, singletonList("001"));
// ...
map.put(979, singletonList("001"));
 ```

So when we're looping through that map we would overwrite the `001` key with, `800` then `808` then `870` etc.  So while the end result is OK because we do `map.remove("001")` I am now worried about our map inversion technique.

Are there any other codes that appear multiple times that we would be overwriting?  If so isn't that almost definitely a problem?

Reply:
One way to find out and make sure it never happens:

```java
for (String region : regions) {
    if (map.containsKey("region")) {
      throw new IllegalStateException("Two country codes for region: " + region):
    }
    map.put(region, code);
}

```

Reply:
Oh shucks! Do you want to go back to the hardcoded version? For now, I've gone with your second comment.